Code/Tools/AssetProcessor/native/unittests/AssetProcessorManagerUnitTests.cpp

```diff
@@ -463,9 +467,8 @@ namespace AssetProcessor
         {
             UNIT_TEST_EXPECT_TRUE(processResults[checkIdx].m_jobEntry.m_computedFingerprint != 0);
             UNIT_TEST_EXPECT_TRUE(processResults[checkIdx].m_jobEntry.m_jobRunKey != 0);
-            UNIT_TEST_EXPECT_TRUE(processResults[checkIdx].m_jobEntry.m_watchFolderPath == AssetUtilities::NormalizeFilePath(watchFolderPath));
-            UNIT_TEST_EXPECT_TRUE(processResults[checkIdx].m_jobEntry.m_pathRelativeToWatchFolder == "uniquefile.txt");
-            UNIT_TEST_EXPECT_TRUE(processResults[checkIdx].m_jobEntry.m_databaseSourceName == "uniquefile.txt");
+            UNIT_TEST_EXPECT_TRUE(QString(processResults[checkIdx].m_jobEntry.m_sourceAssetReference.ScanfolderPath().c_str()) == AssetUtilities::NormalizeFilePath(watchFolderPath));
```

Comment:
Just a heads up that I have another ongoing PR (https://github.com/o3de/o3de/pull/12607) for migrating this test to the modern test framework as well. Not sure which PR will go in first and it may require either of us to resolve the conflict.